### PR TITLE
Adds csminit and csmpt as prototype CSM support

### PR DESCRIPTION
**Pull request by jessemapel**
## Description

This adds two apps:

- csminit attaches a CSM model state to a cube
- csmpt uses an attached CSM model stat to instantiate a CSM model and then run imageToGround

These two demonstrate rudimentary support for CSM in ISIS and are being used to test out the plugin loading and ISD/state loading schemes.

## Related Issue

Continuing CSM in ISIS work

## Motivation and Context

These apps exercise the CSM API in ISIS and demonstrate the logic required to get and use a CSM model. Eventually, much of this code will be housed elsewhere in the code base.

## How Has This Been Tested?

Tested locally with usgscsm and CTX images.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/base/objs/StringBlob/StringBlob.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#ifndef StringBlob_h
+#define StringBlob_h
+
+/**
```
Comment (krlberry) [inline isis/src/base/objs/StringBlob/StringBlob.h]: This is obviously my own fault, but eventually probably needs the new standard text to be pasted at the top of all ISIS files.

File: isis/src/base/objs/StringBlob/StringBlob.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
+/**
```
Comment (krlberry) [inline isis/src/base/objs/StringBlob/StringBlob.cpp]: This is obviously my own fault, but eventually probably needs the new standard text to be pasted at the top of all ISIS files.

File: isis/src/base/apps/csminit/csminit.cpp [hunk 3]
```diff
@@ -0,0 +1,215 @@
+#include "csminit.h"
+
+#include <QList>
+#include <QString>
+#include <QStringList>
+#include <QLibrary>
```
Comment (krlberry) [inline isis/src/base/apps/csminit/csminit.cpp]: Can get rid of this now.

File: isis/src/base/apps/csmpt/csmpt.cpp [hunk 4]
```diff
@@ -0,0 +1,110 @@
+#include "csmpt.h"
+
+#include <QList>
+#include <QString>
+#include <QStringList>
+#include <QLibrary>
```
Comment (krlberry) [inline isis/src/base/apps/csmpt/csmpt.cpp]: Can get rid of this now.

File: isis/src/base/apps/csminit/csminit.xml [hunk 5]
```diff
@@ -0,0 +1,110 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<application name="csminit" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://isis.astrogeology.usgs.gov/Schemas/Application/application.xsd">
+  <brief>
+    Attach a CSM Model state to a cube
+  </brief>
+
+  <description>
+    <p>
+      This program searches takes an ISD or state string and attempts to create
```
Comment (krlberry) [inline isis/src/base/apps/csminit/csminit.xml]: remove "searches"

File: isis/src/base/apps/csminit/csminit.xml [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+      and/or plugin name.
+    </p>
+  </description>
+
+  <category>
+    <categoryItem>Cameras</categoryItem>
+  </category>
+
+  <history>
+    <change name="Jesse Mapel" date="2020-1-18">
```
Comment (krlberry) [inline isis/src/base/apps/csminit/csminit.xml]: Date is incorrect unless you were really planning ahead!

**Review by krlberry — APPROVED** [review]
Some minor comments.

**Comment by jessemapel** [comment]
Don't merge this yet, I'll address Kristin's comments after I get the spiceinit PR up.

**Review by krlberry — APPROVED** [review]
:+1: Looks good!

**Pull request merged**